src/__swaps__/screens/Swap/components/TokenList/TokenToBuyList.tsx

```diff
@@ -97,6 +98,13 @@ export const TokenToBuyList = () => {
   const { internalSelectedInputAsset, internalSelectedOutputAsset, isFetching, isQuoteStale, outputProgress, setAsset } = useSwapContext();
   const { results: sections, isLoading } = useSearchCurrencyLists();
 
+  const [supportedChainsBooleanMap] = useState(
```

Comment:
any reason not to use `useRef` here?

Reply:
I think it's mostly a semantic choice, seems a bit cleaner to use `useState` since you can omit the setter, making it closer to naturally read only, and with `useState` you don't need to deal with `.current`.

Reply:
Sounds reasonable to me. Was just curious!